frontend/src/components/HardwareDialog.tsx

```diff
@@ -1,50 +1,97 @@
 import { useState, useRef, useEffect } from "react";
 
-interface HardwareDialogProps {
-    isOpen: boolean; // Determines if the dialog is open
-    onClose: () => void; // Function to close the dialog
-    onSubmit: (action: "checkOut" | "checkIn", hardwareId: string, quantity: number) => void; // Function to handle submit
+// Define an enum for action types
+enum ActionType {
+    CheckOut = "checkOut",
+    CheckIn = "checkIn",
```

Comment:
I think its just easier to duplicate the code rather than adding a bunch of complicated logic and state